#**Practical-8: Descriptive Statistics**
**Name: Preeti Sahani**

In [ ]:
import pandas as pd

df=pd.read_csv('/content/stats.csv')
df

Name  Salary Country
0        Dan   40000     USA
1  Elizabeth   32000  Brazil
2        Jon   45000   Italy
3      Maria   54000     USA
4       Mark   72000     USA
5       Bill   62000  Brazil
6       Jess   92000   Italy
7      Julia   55000     USA
8       Jeff   35000   Italy
9        Ben   48000  Brazil

#**Measure of Central Tendency**

In [ ]:
# Mean Salary
mean1=df['Salary'].mean()
mean1

53500.0

In [ ]:
# Sum of Salaries
sum1=df['Salary'].sum()
sum1

535000

In [ ]:
# Maximum Salary
max1=df['Salary'].max()
max1

92000

In [ ]:
# Minimum Salary
min1=df['Salary'].min()
min1

32000

In [ ]:
# Total Count
count1=df['Salary'].count()
count1

10

In [ ]:
# Median
median=df['Salary'].median()
median

51000.0

In [ ]:
# Mode
mode1=df['Salary'].mode()
mode1

0    32000
1    35000
2    40000
3    45000
4    48000
5    54000
6    55000
7    62000
8    72000
9    92000
dtype: int64

In [ ]:
countrywise_sum=df.groupby(['Country']).sum()
countrywise_sum

Salary
Country        
Brazil   142000
Italy    172000
USA      221000

In [ ]:
countrywise_count=df.groupby(['Country']).count()
countrywise_count

Name  Salary
Country              
Brazil      3       3
Italy       3       3
USA         4       4

#**Measure of Variability**

In [ ]:
# Variance of salaries
var1=df['Salary'].var()
var1

332055555.5555556

In [ ]:
# Standard deviation
std1=df['Salary'].std()
std1

18222.391598128816

#**Measure of Symmetry**

In [ ]:
skew1=df.skew(axis=0, skipna=True)
skew1

Salary    1.021551
dtype: float64

**The Skewness is positive so x will have right side tail.**

#**Covariance and Correlation**

In [ ]:
import pandas as pd
bw=pd.read_csv('/content/BirthWeight.csv')
bw.head()

Infant ID  Gestational Age (Weeks)  Birth Weight (Grams)
0          1                     34.7                  1895
1          2                     36.0                  2030
2          3                     29.3                  1440
3          4                     40.1                  2835
4          5                     35.7                  3090

In [ ]:
bw.set_index('Infant ID', inplace=True)
bw.head()

Gestational Age (Weeks)  Birth Weight (Grams)
Infant ID                                               
1                             34.7                  1895
2                             36.0                  2030
3                             29.3                  1440
4                             40.1                  2835
5                             35.7                  3090

In [ ]:
print(bw.cov())

print(bw.corr(method="pearson"))

                         Gestational Age (Weeks)  Birth Weight (Grams)
Gestational Age (Weeks)                 9.963824              1798.025
Birth Weight (Grams)                 1798.025000            485478.750
                         Gestational Age (Weeks)  Birth Weight (Grams)
Gestational Age (Weeks)                 1.000000              0.817519
Birth Weight (Grams)                    0.817519              1.000000


**Covariance indicates that there is correlation exists between two Correlation coefficient of 0.818 indicates the relation between two variable is strong.**

In [1]:
#importing required libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import skew
from scipy.stats import kurtosis

In [ ]:
pd.set_option("display.max_columns",None)
pd.options.display.float_format = "{:,.2f}".format

**Format : A data frame with 53940 rows and 10 variables**

**Description : A dataset containing the prices and other attributes of almost 54,000 diamonds.**

**The variables are as follows:**

price: price in US dollars ($326--$18,823)

carat: weight of the diamond (0.2--5.01)

cut: quality of the cut (Fair, Good, Very Good, Premium, Ideal)

colour: diamond colour, from J (worst) to D (best)

clarity: a measurement of how clear the diamond is (IF (best), VVS1, VVS2, VS1, VS2, SI1, SI2, I1 (worst) )

popularity: popularity of this specs (Good, Fair, Poor)

x: length in mm (0--10.74)

y: width in mm (0--58.9)

z: depth in mm (0--31.8)

depth: total depth percentage = z / mean(x,y) = 2 * z /(x+y) (43--79) 

table: width of top of diamondrelative to widest point (43--95)

In [3]:
# reading data from csv file
xls = pd.read_csv("/content/diamonds.csv")
xls.head()

id  carat      cut color clarity  depth  table  price     x     y     z
0   1   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98  2.43
1   2   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84  2.31
2   3   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07  2.31
3   4   0.29  Premium     I     VS2   62.4   58.0    334  4.20  4.23  2.63
4   5   0.31     Good     J     SI2   63.3   58.0    335  4.34  4.35  2.75

In [4]:
des_df = xls.drop(['id'],axis= 1) #drop id colums
for col in des_df:            #drop all alpha-numeric colms
    if des_df[col].dtype =='object':
        des_df = des_df.drop([col],axis = 1)
        
des_r = des_df.describe()
des_df = des_r.rename(index={'50%':'median/50%'})
des_r

carat         depth  ...             y             z
count  53940.000000  53940.000000  ...  53940.000000  53940.000000
mean       0.797940     61.749405  ...      5.734526      3.538734
std        0.474011      1.432621  ...      1.142135      0.705699
min        0.200000     43.000000  ...      0.000000      0.000000
25%        0.400000     61.000000  ...      4.720000      2.910000
50%        0.700000     61.800000  ...      5.710000      3.530000
75%        1.040000     62.500000  ...      6.540000      4.040000
max        5.010000     79.000000  ...     58.900000     31.800000

[8 rows x 7 columns]

In [5]:
var_r = des_df.var() #Calculating variance seperately

varlist = []
for col in des_df.columns:
    if des_df[col].dtype == 'object':
        continue
    varlist.append(round(des_df[col],5))

df = pd.DataFrame([varlist],columns=des_r.columns,index=['var'])
mct = des_r.append(df) #Adding var to describe result
mct

carat  ...                                                  z
count                                              53940  ...                                              53940
mean                                             0.79794  ...                                            3.53873
std                                             0.474011  ...                                           0.705699
min                                                  0.2  ...                                                  0
25%                                                  0.4  ...                                               2.91
50%                                                  0.7  ...                                               3.53
75%                                                 1.04  ...                                               4.04
max                                                 5.01  ...                                               31.8
var    count         53940.00000
mean              0....  ...  count         53940.00000
mean              3....

[9 rows x 7 columns]